File: leetcode_75/Array_String/python/1768.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        result = ""
        l1 = len(word1)
        l2 = len(word2)
        ptr1 = ptr2 = 0
        while ptr1 < l1 or ptr2 < l2:
            if ptr1 < l1:
                result += word1[ptr1]
                ptr1 += 1
            if ptr2 < l2:
                result += word2[ptr2]
                ptr2 += 1
        return result

File: leetcode_75/Array_String/js/1768.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function (word1, word2) {
    let result = []
    let startLength = Math.max(word1.length, word2.length)
    let i = 0
    while (i < startLength) {
        result.push(word1[i])
        result.push(word2[i])
        i++
    }

    if (i < word1.length) {
        result.push(word1[i++])
    }
    if (i < word2.length) {
        result.push(word2[i++])
    }

    return result.join("")
};